connectors/http/polling/src/main/java/io/camunda/connector/http/polling/task/PollingOperateTask.java

```diff
@@ -0,0 +1,92 @@
+
+import io.camunda.connector.api.inbound.InboundIntermediateConnectorContext;
+import io.camunda.connector.api.inbound.ProcessInstanceContext;
+import io.camunda.connector.http.base.services.HttpService;
+import io.camunda.connector.http.polling.service.HttpPollingSharedExecutorService;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PollingOperateTask is responsible for handling HTTP polling operations. Each instance of this
+ * class is used for one flowNode and manages polling across all its corresponding processInstances.
+ */
+public class PollingOperateTask extends AbstractScheduledTask {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PollingOperateTask.class);
+
+  private final InboundIntermediateConnectorContext context;
+  private final HttpService httpService;
+  private final HttpPollingSharedExecutorService executorService;
+  private final Duration httpRequestInterval;
+  private final Set<String> runningHttpRequestTaskIds;
+
+  public PollingOperateTask(
+      final String id,
+      final InboundIntermediateConnectorContext context,
+      final HttpService httpService,
+      final HttpPollingSharedExecutorService executorService,
+      final Duration httpRequestInterval) {
+    super(id);
+    this.context = context;
+    this.httpService = httpService;
+    this.executorService = executorService;
+    this.httpRequestInterval = httpRequestInterval;
+    this.runningHttpRequestTaskIds = new HashSet<>();
+  }
+
+  @Override
+  public void run() {
+    try {
+      List<ProcessInstanceContext> processInstanceContexts = context.getProcessInstanceContexts();
+      if (processInstanceContexts != null) {
+        removeInactiveTasks(processInstanceContexts);
+        scheduleNewTasks(processInstanceContexts);
+      }
+    } catch (Exception e) {
+      LOGGER.error("An error occurred: {}", e.getMessage(), e);
+    }
+  }
+
+  private void removeInactiveTasks(final List<ProcessInstanceContext> processInstanceContexts) {
+    List<String> activeTasks =
+        processInstanceContexts.stream().map(this::getRequestTaskKey).toList();
+
+    List<String> inactiveTasks =
+        runningHttpRequestTaskIds.stream().filter(key -> !activeTasks.contains(key)).toList();
+
+    inactiveTasks.forEach(executorService::cancelTask);
+  }
+
+  private void scheduleNewTasks(List<ProcessInstanceContext> contexts) {
+    contexts.forEach(
+        processInstanceContext -> {
+          String taskKey = getRequestTaskKey(processInstanceContext);
+
+          if (!executorService.containsTask(taskKey)) {
+            HttpRequestTask newTask =
+                new HttpRequestTask(taskKey, httpService, this.context::correlate);
+            newTask.setProcessInstanceContext(processInstanceContext);
+
+            runningHttpRequestTaskIds.add(taskKey);
```

Comment:
This is not needed as discussed yesterday :)

Reply:
This should be tracked by the executor service. Just add a method: `executorService.getTaskIds()`;

Reply:
Just schedule the operate polling task with the same executorservice but without this task abstraction.

Reply:
Just use: `this.executorService.schedule...`